skyplane/obj_store/file_system_interface.py

```diff
@@ -0,0 +1,65 @@
+import os
+from dataclasses import dataclass
+from typing import Iterator, List, Optional, Tuple
+from skyplane.obj_store.object_store_interface import ObjectStoreObject
+
+
+@dataclass
+class LocalFile:
+    """Defines file on local node."""
+
+    path: str
+    size: Optional[int] = None
+    last_modified: Optional[str] = None
+    file_format: Optional[str] = None
+
+    @property
+    def exists(self):
+        raise NotImplementedError()
+
+    def write_permissions(self):
+        raise NotImplementedError()
+
+    def real_path(self):
+        raise NotImplementedError()
+
+
+class FileSystemInterface:
```

Comment:
Should this align more closely with a POSIX file system API i.e. open, close, read etc.

Reply:
This is a slightly longer design discussion. Let's discuss it in today's call.